### PR TITLE
Add dark mode transformation in 'Editor.replaceNode()' 

### PR DESCRIPTION
**Pull request by Rain-Zheng**
When replacing an existing node with a new node in dark mode, `Editor.replaceNode()` doesn't transform the new node to dark mode. This PR transforms it like 'Editor.insertNode()' does.

### CODE REVIEW

**Comment by Rain-Zheng** [comment]
@JiuqingSong Could you take a look at it? Thanks!

File: packages/roosterjs-editor-core/lib/editor/Editor.ts [hunk 1]
```diff
@@ -209,9 +209,23 @@ export default class Editor {
      * @returns true if node is replaced. Otherwise false
      */
     public replaceNode(existingNode: Node, toNode: Node): boolean {
+        // Transform the new node to replace to
+        const darkModeOptions = this.getDarkModeOptions();
+        const darkModeTransform = this.isDarkMode()
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-core/lib/editor/Editor.ts:214]: This changes the existing behavior. Some customer may rely on the existing behavior to not convert replaced node to dark mode. 
Consider add a parameter to control this behavior

Comment (Rain-Zheng) [reply packages/roosterjs-editor-core/lib/editor/Editor.ts]: Got it, a param has been added.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "roosterjs",
-    "version": "7.15.2",
+    "version": "7.15.3",
```
Comment (JiuqingSong) [inline package.json:3]: New parameter added, use 7.16.0 instead

File: packages/roosterjs-editor-core/lib/editor/Editor.ts [hunk 3]
```diff
@@ -206,12 +206,30 @@ export default class Editor {
      * Replace a node in editor content with another node
      * @param existingNode The existing node to be replaced
      * @param new node to replace to
+     * @param darkMode (optional) Whether to transform new node to dark mode. Default is false
      * @returns true if node is replaced. Otherwise false
      */
-    public replaceNode(existingNode: Node, toNode: Node): boolean {
+    public replaceNode(existingNode: Node, toNode: Node, darkMode?: boolean): boolean {
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-core/lib/editor/Editor.ts:212]: suggest, rename "darkMode" to "transformColorForDarkMode"

Comment (Rain-Zheng) [reply packages/roosterjs-editor-core/lib/editor/Editor.ts]: Updated. Thanks!

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**